development/performance/update_params_benchmark/time_update_parameters.py

```diff
@@ -0,0 +1,67 @@
+from simtk import openmm, unit
+from simtk.openmm import app
+from openmmtools.testsystems import WaterBox
+
+"""
+Script to time the functions 'setParticleParameters' and 'updateParametersInContext' in OpenMM by partially deleting
+and re-instating a water molecule in a box of water molecules.
+"""
+
+def make_water_system(length=20.0, cutoff=9.0, platform='CUDA', equil_steps=10):
+    """
+    Create a box of TIP3P water and return the compenents of the OpenMM system.
+    """
+    wbox = WaterBox(box_edge=length * unit.angstrom, cutoff=cutoff * unit.angstrom, nonbondedMethod=app.PME)
+    integrator = openmm.LangevinIntegrator(300*unit.kelvin, 1/unit.picosecond, 0.002*unit.picoseconds)
+    if platform == 'CUDA':
+        openmm_platform = openmm.Platform.getPlatformByName('CUDA')
+        properties = {'CudaPrecision': 'mixed'}
+        context = openmm.Context(wbox.system, integrator, openmm_platform, properties)
+    elif platform == 'OpenCL':
+        openmm_platform = openmm.Platform.getPlatformByName('OpenCL')
+        properties = {'OpenCLPrecision': 'mixed'}
+        context = openmm.Context(wbox.system, integrator, openmm_platform, properties)
+    else:
+        context = openmm.Context(wbox.system, integrator)
+    context.setPositions(wbox.positions)
+    context.setVelocitiesToTemperature(300*unit.kelvin)
+    integrator.step(equil_steps)
+    return context, integrator, wbox
+
+def switch_water_force(force, context,frac=0.9):
+    """
+    Switch the non-bonded parameters of the first TIP3P water molecule to some specified fraction of the default values.
+    """
+    force.setParticleParameters(0, charge=-0.834*frac, sigma=0.3150752406575124*frac, epsilon=0.635968*frac)
+    force.setParticleParameters(1, charge=0.417*frac, sigma=0, epsilon=1*frac)
+    force.setParticleParameters(2, charge=0.417*frac, sigma=0, epsilon=1*frac)
+    force.updateParametersInContext(context)
+
+def run_timing(length=20.0, niterations=20, platform='CPU'):
```

Comment:
We certainly did have a scripts for comparing dynamics with and without parameter updates (for the record, those scripts also show the recent slowdown in the code). The script `time_update_parameters.py` is indeed based on those, although is more minimal as no dynamics are propagated. The new scripts grants more flexibility and ease of automation as I've pre-built different `conda` enviroments, as well as made the the `openmm` platform, water box size, and number of iterations settable from the command line.

The PR shows that the slowdown in `updateParametersInContext` exists for a number of different platforms and system sizes.